pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -134,6 +146,183 @@ func (m *KubeIPResolverInstance) EnrichEvent(ev any) error {
 	return nil
 }
 
+func (k *KubeIPResolver) GlobalParams() api.Params {
+	return nil
+}
+
+func (k *KubeIPResolver) InstanceParams() api.Params {
+	return nil
+}
+
+type endpointAccessors struct {
+	root            datasource.FieldAccessor
+	subK8sKind      datasource.FieldAccessor
+	subK8sName      datasource.FieldAccessor
+	subK8sNamespace datasource.FieldAccessor
+	subK8sLabels    datasource.FieldAccessor
+}
+
+func (k *KubeIPResolver) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	k8sInventory, err := common.GetK8sInventoryCache()
+	if err != nil {
+		return nil, fmt.Errorf("creating k8s inventory cache: %w", err)
+	}
+	inst := &KubeIPResolverInstance{
+		k8sInventory: k8sInventory,
+	}
+
+	logger := gadgetCtx.Logger()
+	epAccessors := make(map[datasource.DataSource][]endpointAccessors)
+	for _, ds := range gadgetCtx.GetDataSources() {
+		logger.Debugf("KubeIPResolverOperator inspecting datasource %q", ds.Name())
+
+		endpoints := ds.GetFieldsWithTag("type:" + endpointL4Type)
+		if len(endpoints) > 0 {
+			logger.Debugf("> found %d endpoint fields", len(endpoints))
+			for _, ep := range endpoints {
+				k8sSubAcc, err := ep.AddSubField("k8s", api.Kind_Invalid, datasource.WithFlags(datasource.FieldFlagEmpty))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "k8s", err)
+				}
+				k8sKindAcc, err := k8sSubAcc.AddSubField("kind", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
```

Comment:
Add some columns annotations here to set the width and max width to 3 as it can be just "pod" or "svc".

Reply:
It seems if I set the width to 3 the column header is `...` which doesn't look nice!

Reply:
Ahhh, good point. So, it seems like the best we can do is to set max width to 12 (len of `{SRC|DST}.K8S.KIND`). WDYT?